pkg/maps/lbmap/lbmap.go

```diff
@@ -348,15 +247,55 @@ func updateMasterService(fe ServiceKey, nbackends int, nonZeroWeights uint16) er
 	return updateService(fe, zeroValue)
 }
 
-// UpdateService adds or updates the given service in the bpf maps
-func UpdateService(fe ServiceKey, backends []ServiceValue, addRevNAT bool, revNATID int) error {
+// UpdateService adds or updates the given service in the bpf maps (in both -
+// legacy and v2).
+func UpdateService(fe ServiceKey, backends []ServiceValue,
```

Comment:
I thought about this, but considering that the code got more complex and the legacy updates will go in the nearest future I didn't want to introduce more functions which IMHO would make the function less readable. I could change this though if you think otherwise.

Reply:
For any function that's much longer than my screen, I struggle to keep everything in my head, and I find it more difficult to determine variable usage and potential side-effects. I won't press on this, but the longer the function is, the more incentive I have to avoid interacting with it :-).

Reply:
OK;-)

I gave a try to make it more readable by splitting into multiple smaller functions.